pkg/kubelet/dockershim/checkpoint_store.go

```diff
@@ -76,7 +76,10 @@ func writeFileAndSync(filename string, data []byte, perm os.FileMode) error {
 	if err == nil && n < len(data) {
 		err = io.ErrShortWrite
 	}
-	f.Sync()
+	if err == nil {
+		// Only sync if the Write completed successfully.
+		err = f.Sync()
```

Comment:
Is there a reason for hiding old errors behind new errors? If it's so control flow makes it to `f.Close()`, perhaps `defer` is better?

Reply:
This function surfaces the errors in the following order:
 1. write error
 2. sync error
 3. close error

If (1) is non-nil, return (1). If (2) is non-nil, return (2), etc.
This follows the existing flow and convention in `ioutil.WriteFile()`

Reply:
Ah, I see. SG.